src/main/java/io/split/android/client/storage/mysegments/MySegmentsStorageContainerImpl.java

```diff
@@ -0,0 +1,44 @@
+package io.split.android.client.storage.mysegments;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import androidx.annotation.NonNull;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class MySegmentsStorageContainerImpl implements MySegmentsStorageContainer {
+
+    private final ConcurrentMap<String, MySegmentsStorage> mStorageMap = new ConcurrentHashMap<>();
+    private final PersistentMySegmentsStorage mPersistentMySegmentsStorage;
+    private final Object lock = new Object();
+
+    public MySegmentsStorageContainerImpl(@NonNull PersistentMySegmentsStorage persistentMySegmentsStorage) {
+        mPersistentMySegmentsStorage = checkNotNull(persistentMySegmentsStorage);
+    }
+
+    @NonNull
+    @Override
+    public MySegmentsStorage getStorageForKey(String matchingKey) {
```

Comment:
Is this a naming convention we have through these modules or from Android? otherwise, have you considered a name that reflects that if not there it'll create it? 

Just a curiosity comment.